### PR TITLE
New : integration gpec, source code

### PR DESCRIPTION
**Pull request by atm-gauthier**

### CODE REVIEW

**Comment by atm-gauthier** [comment]
Hi @eldy, i also did hrm_position > hrm_job_user adaptations on source code but i didn't rename concerned object and files because it would be a very long job.
Is it ok for you ?

File: htdocs/hrm/lib/hrm.lib.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * \ingroup hr
+ * \brief   Library files with common functions for Workstation
+ */
+
+/**
+ * Prepare admin pages header
+ *
+ * @return array
+ */
+function hrmAdminPrepareHead()
```
Comment (eldy) [inline htdocs/hrm/lib/hrm.lib.php:33]: We are loosing the tab that link to the existing page to setup the "department" of a company:
"admin_establishment.php" ?

Comment (atm-gauthier) [reply htdocs/hrm/lib/hrm.lib.php:33]: Hi @eldy, I will have a look at this on Monday. Have a nice week-end.

Comment (atm-gauthier) [reply htdocs/hrm/lib/hrm.lib.php:33]: Hi @eldy, i just restore link to hrm establihsment tab

**Comment by atm-gauthier** [comment]
Hi @eldy, i have a problem with travis : 

I get this feedback : Found non escaped string in building of a sql request (case 3) in hrm/class/evaluation.class.php: $sql .= ' AND ('.implode(' '.$filtermode.' ', $sqlwhere).')';

But if i add $this->db->escape on value i get an error because for my request in skill_tab.php i need to use fetch_all with objecttype='job'

AND for postgresql compatibility, i needed to use simple quotes and not double quotes.

Then if i add $this->db->escape on $sqlwhere i will get something like : 

image

and it breaks my query...

What can i do ?

It seems that some travis feedback are not up to date with what modulebuilder creates in class

**Comment by eldy** [comment]
Replace  
$sql .= ' AND ('.implode(' '.$filtermode.' ', $sqlwhere).')';
with
$sql .= " AND (".implode(" ".$filtermode." ", $sqlwhere).")";

To forge sql request we must use " for php string and ' is used for sql string. To allow detection of sql injection we must follow strictly this rule.

**Comment by atm-gauthier** [comment]
@eldy i'm working to correct this and the same problem in a lot of file.
But i think module builder should be corrected too because a lot of files i'm fixing come from module builder use.

**Comment by atm-gauthier** [comment]
@eldy you have a parse error in develop : 

image

It is blocking for travis build

**Comment by atm-gauthier** [comment]
i did a pr https://github.com/Dolibarr/dolibarr/pull/18918

**Comment by atm-gauthier** [comment]
Hello @eldy, now all checks have passed for this PR.
You will see that besides hrm files, i had to add some modifications on several standard files, to help you i list them here :
htdocs/core/lib/company.lib.php
htdocs/user/class/user.class.php

If you have some questions don't hesitate

**Pull request merged**